govcd/upload.go

```diff
@@ -105,19 +105,22 @@ func uploadFile(client *Client, filePath string, uDetails uploadDetails) (int64,
 }
 
 // Create Request with right headers and range settings. Support multi part file upload.
+// client - client for requests
 // requestUrl - upload url
 // filePart - bytes to upload
 // offset - how much is uploaded
 // filePartSize - how much bytes will be uploaded
 // fileSizeToUpload - final file size
-func newFileUploadRequest(requestUrl string, filePart []byte, offset, filePartSize, fileSizeToUpload int64) (*http.Request, error) {
+func newFileUploadRequest(client *Client, requestUrl string, filePart []byte, offset, filePartSize, fileSizeToUpload int64) (*http.Request, error) {
 	util.Logger.Printf("[TRACE] Creating file upload request: %s, %v, %v, %v \n", requestUrl, offset, filePartSize, fileSizeToUpload)
 
-	uploadReq, err := http.NewRequest(http.MethodPut, requestUrl, bytes.NewReader(filePart))
+	parsedRequestURL, err := url.ParseRequestURI(requestUrl)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error decoding vdc response: %s", err)
 	}
 
+	uploadReq := client.NewRequestWitNotEncodedParams(nil, nil, http.MethodPut, *parsedRequestURL, bytes.NewReader(filePart))
```

Comment:
Just noticed the typo in the func name "Wit", but probably it's better not to touch this public function at this late point in the release.

Reply:
Oh nice, spot :) We can live with that for a while :-D this internal public function